src/DynamoRevit/ViewModel/RevitWatchHandler.cs

```diff
@@ -46,28 +47,28 @@ private WatchViewModel ProcessThing(Element element, ProtoCore.RuntimeCore runti
         }
 
         //If no dispatch target is found, then invoke base watch handler.
-        private WatchViewModel ProcessThing(object obj, ProtoCore.RuntimeCore runtimeCore, string tag, bool showRawData, WatchHandlerCallback callback)
+        private WatchViewModel ProcessThing(object obj, List<string> preferredDictionaryOrdering, ProtoCore.RuntimeCore runtimeCore, string tag, bool showRawData, WatchHandlerCallback callback)
         {
-            return baseHandler.Process(obj, runtimeCore, tag, showRawData, callback);
+            return baseHandler.Process(obj, preferredDictionaryOrdering, runtimeCore, tag, showRawData, callback);
         }
 
-        private WatchViewModel ProcessThing(MirrorData data, ProtoCore.RuntimeCore runtimeCore, string tag, bool showRawData, WatchHandlerCallback callback)
+        private WatchViewModel ProcessThing(MirrorData data, List<string> preferredDictionaryOrdering, ProtoCore.RuntimeCore runtimeCore, string tag, bool showRawData, WatchHandlerCallback callback)
         {
             try
             {
-                return baseHandler.Process(data, runtimeCore, tag, showRawData, callback);
+                return baseHandler.Process(data, preferredDictionaryOrdering, runtimeCore, tag, showRawData, callback);
             }
             catch (Exception)
             {
-                return callback(data.Data, runtimeCore, tag, showRawData);
+                return callback(data.Data, preferredDictionaryOrdering, runtimeCore, tag, showRawData);
             }
         }
 
-        public WatchViewModel Process(dynamic value, ProtoCore.RuntimeCore runtimeCore, string tag, bool showRawData, WatchHandlerCallback callback)
+        public WatchViewModel Process(dynamic value, IEnumerable<string> preferredDictionaryOrdering, ProtoCore.RuntimeCore runtimeCore, string tag, bool showRawData, WatchHandlerCallback callback)
         {
```

Comment:
just curious why you pass an `IEnumerable` here and call `toList()` while just passing `List` elsewhere?

Reply:
It's generally better practice to use `IEnumerable` (interfaces) in public facing API's so that if the concrete implementation changes from List to say Dictionary or something else, then client code need not be updated. However in this case it's not so useful. Usually private methods have concrete types like List, etc.